File: Easy/1 - Two Sum/1Soln.js
'use strict'

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
    const hashNums = {}

    // Generate hash array
    for (let i = 0; i < nums.length; i++) {
        hashNums[nums[i]] = i
    }

    // Perform check
    let diff
    for (let i = 0; i < nums.length; i++) {
        diff = target - nums[i]
        if (hashNums[diff] !== undefined && hashNums[diff] !== i) {
            return [i, hashNums[diff]]
        }
    }

    return []
}

File: Easy/1 - Two Sum/1Soln.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_nums = {}

        # Create hash
        for i, val in enumerate(nums):
            hash_nums[val] = i

        # Check hash
        for i, val in enumerate(nums):
            diff = target - val
            if diff in hash_nums.keys() and hash_nums[diff] != i:
                return [i, hash_nums[diff]]

        return []
        
